### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by breedx-splk**
Downstream projects are getting hit with protobuf updates that can't be merged. Hoping this might help.

grpc 1.56.1 -> 1.65.1
protobuf 3.23.4 -> 4.27.2

also bumped a couple of github actions.

### CODE REVIEW

**Comment by breedx-splk** [comment]
* Reference: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10845
* Reference: https://github.com/open-telemetry/opentelemetry-java/pull/6295
* Reference: https://github.com/open-telemetry/opentelemetry-java-examples/pull/395

**Review by trask — APPROVED** [review]

**Comment by jack-berg** [comment]
Ran the build locally to confirm this works. Unfortunately, we don't have project tooling at the moment to do a patch release. The release build accepts a released version of `opentelemetry-proto` as an input, builds against it, and releases with the same version. If we merge this and try to re-run the release build with `1.3.2` we'll get an error.

We can either:
- Wait for the next `opentelemetry-proto` release
- Update the project tooling to be able to sometimes append a suffix to the release version string to represent a patch

**Review by jack-berg — APPROVED** [review]

**Comment by trask** [comment]
seems ok to wait (btw thx @breedx-splk for this, I wasn't clear what we needed to do to unblock those dependency updates)

**Comment by breedx-splk** [comment]
@jack-berg I'm fine waiting for the next release of `opentelemetry-proto`, I think that's fine. Judging by the release history over there, it seems like a July version should happen shortly anyway. If the release builds are triggered manually we can merge this tho right?

**Comment by breedx-splk** [comment]
> seems ok to wait (btw thx @breedx-splk for this, I wasn't clear what we needed to do to unblock those dependency updates)

There may still be more, but this is probably part of it.

**Comment by jack-berg** [comment]
> If the release builds are triggered manually we can merge this tho right?

Yes.

**Pull request merged**
